src/main/java/com/imgix/URLHelper.java

```diff
@@ -184,6 +177,22 @@ public static String encodeURIComponent(String s) {
     return result;
   }
 
+  public static String encodeURI(String s) {
+    String result =
+        s.replaceAll("\\+", "%2B")
+            .replaceAll("\\:", "%3A")
+            .replaceAll("\\?", "%3F")
+            .replaceAll("\\#", "%23")
+            .replaceAll(" ", "%20")
+            .replaceAll("\\%21", "!")
+            .replaceAll("\\%27", "'")
+            .replaceAll("\\%28", "(")
+            .replaceAll("\\%29", ")")
+            .replaceAll("\\%7E", "~");
+
+    return result.replace("\\", "");
+  }
```

Comment:
Kept trying to think of a way to do this with some iteration of Java for loop syntax and or case statements.

In almost every case the string has to be converted into an array and then back into a string. Seems like the loops we save ourselves might also introduce more opportunities to incorrectly overwrite the path string. So leaving this as is with the hopes a better solution comes to mind.

Reply:
Addressed in the stacked PR.